flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphCoLocationRestartTest.java

```diff
@@ -51,6 +51,9 @@
 
 	@Test
 	public void testConstraintsAfterRestart() throws Exception {
+
```

Comment:
I think in the `ExecutionGraphCoLocationRestartTest` it is not necessary to use a proper `SlotPool`. Instead we could use a `TestingSlotProvider` which always returns completed futures. Then we could use `TestComponentMainThreadExecutor#forMainThread` and would not need to wrap the methods in the `execute` calls.

Reply:
I think this might be not as a easy. It seems that some more slot pool functionality is required in this test to fulfill the asserts in `validateConstraints(...)`